packages/connect-core/src/entities/Organization.ts

```diff
@@ -25,35 +26,68 @@ import { ConnectorInterface } from '../connections/ConnectorInterface'
 // [ ] Events...
 
 export default class Organization {
-  #address: string
+  readonly location: string
+  #address?: string
   #provider: ethers.providers.Provider
+  #connected: boolean
 
   private _connector: ConnectorInterface
 
   constructor(
-    address: string,
+    location: string,
     connector: ConnectorInterface,
-    provider?: ethers.providers.Provider,
-    chainId?: number
+    provider: any,
+    network: Networkish
   ) {
-    this.#address = address
-    this.#provider =
-      provider ||
-      new ethers.providers.InfuraProvider(chainId || connector.chainId || 1)
+    this.location = location
+
+    this.#provider = provider
+      ? new ethers.providers.Web3Provider(provider, network)
+      : new ethers.providers.InfuraProvider(network)
```

Comment:
What you think of instead using the `FallbackProvider`? I think we might see a better performance. For example, we can decide to use a weight of 0.5 for our own node and then distribute the other half equally between the API providers (EtherscanProvider, AlchemyProvider, InfuraProvider).

Reply:
Ah yes I was checking it recently, it sounds super interesting! The only thing is that it doesn’t seem to support a network configuration (i.e. a custom ENS address)?

Reply:
Oh maybe is not possible but I thought that would be configurable using the: [FallbackProviderConfig](https://docs.ethers.io/v5/api/providers/other/#FallbackProviderConfig)

Reply:
Oooh I understand now, we pass a configuration for each of them. It should work then, as long as each provider we want also supports custom ENS addresses :+1:

Reply:
At least is how I understood it, but we can give it a shoot for the next release for sure 🙌